src/citrine/ara/columns.py

```diff
@@ -0,0 +1,83 @@
+"""Column definitions for Ara."""
+from typing import Type, Optional, List  # noqa: F401
+from abc import abstractmethod
+
+from citrine._serialization.serializable import Serializable
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization import properties
+
+
+class Column(PolymorphicSerializable['Column']):
+    """A column in the Ara table, defined as some operation on a variable.
+
+    Abstract type that returns the proper type given a serialized dict.
+    """
+
+    @abstractmethod
+    def _attrs(self) -> List[str]:
+        pass  # pragma: no cover
+
+    def __eq__(self, other):
+        try:
+            return all([
+                self.__getattribute__(key) == other.__getattribute__(key) for key in self._attrs()
+            ])
+        except AttributeError:
+            return False
+
+    @classmethod
+    def get_type(cls, data) -> Type[Serializable]:
+        """Return the subtype."""
+        types: List[Type[Serializable]] = [
+            RealMeanColumn, IdentityColumn
+        ]
+        return next(x for x in types if x.type == data["type"])
+
+
+class RealMeanColumn(Serializable['RealMeanColumn'], Column):
+    """Column containing the mean of a real-valued variable.
+
+    Parameters
+    ----------
+    data_source: str
+        short_name of the variable to use when populating the column
+    target_units: optional[str]
+        units to convert the real variable into
```

Comment:
Why are `target_units` optional?  What is the default?

Reply:
They are optional with the default `None`, in which case they are returned with original units.  You'd only expect to use that if you then had the original units as a separate display column.  I can pull that into this minimal example for clarity (note that its already in the backend prototype).

Reply:
I'd think a default of a non-conversion would be a footgun.  Can we have `None` be an option, which behaves as you say, but default be empty string (or functional equivalent), in which case the behavior is grab the output units from the template?  That would make much more sense as a default to me.

Reply:
I'd rather have the "by default grab it from the template" behavior be part of the SDK behavior and not part of the API/python binding.  But I do like that being default behavior in the SDK...

Reply:
I'm all for handling behavior in the SDK, but would feel better if we swapped it to a required argument if the nominally default behavior would be problematic if implemented as such.